app/services/account-manager.js

```diff
@@ -0,0 +1,98 @@
+import Ember from 'ember';
+import config from '../config/environment';
+import { purecloudEnvironment } from '../utils/purecloud-environment';
+
+export default Ember.Service.extend({
+	purecloud: Ember.inject.service('purecloud'),
+	savedAccounts: [],
+	localInitialized: [],
+	selectedAccount: null,
+
+	getSelected: Ember.observer('selectedAccount', function () {
+		let selectedAccount = this.get('selectedAccount');
+		this.get('purecloud').setSelected(selectedAccount);
+	}),
+
+	init() {
+		let storage = window.localStorage;
+		let initializedAccounts = JSON.parse(storage.getItem('initialized'));
+		this.localInitialized = initializedAccounts.accounts;
+		let selectedAccount = JSON.parse(storage.getItem('selectedAccount'));
+		this.setSelected(selectedAccount);
+		let savedAccountsData = JSON.parse(storage.getItem('accounts'));
+		this.savedAccounts = savedAccountsData.accounts;
+	},
+
+	setSelected(account) {
+		let temp = [];
+		for (let i = 0; i < this.localInitialized.length; i++) {
+			this.localInitialized[i].status = false;
+			if (this.localInitialized[i].userId === account.userId) {
+				this.localInitialized[i].status = true;
+			}
+		}
+		temp = [...this.localInitialized];
+		this.set('initialized', temp);
+
+		//store modified initialized accounts in local storage
+		let storedInitialized = JSON.parse(window.localStorage.getItem('initialized'));
+		storedInitialized.accounts = temp;
+		window.localStorage.setItem('initialized', JSON.stringify(storedInitialized));
+
+		//Set account as selected account and save in local storage
+		this.set('selectedAccount', account);
+		window.localStorage.setItem('selectedAccount', JSON.stringify(account));
+	},
+
+	deleteAccount(accountId, token) {
+		let tempAccounts = [];
+		let tempInitialized = [];
+		for (let i = 0; i < this.savedAccounts.length; i++) {
+			if (this.savedAccounts[i].userId === accountId && this.localInitialized[i].userId === accountId) {
+				this.savedAccounts.splice(i, 1);
+				this.localInitialized.splice(i, 1);
+				tempAccounts = [...this.savedAccounts];
+				tempInitialized = [...this.localInitialized];
+			}
+		}
+
+		//Store modified initialized accounts
+		let storedInitialized = JSON.parse(window.localStorage.getItem('initialized'));
+		storedInitialized.accounts = tempInitialized;
+		this.set('initialized', tempInitialized);
+		window.localStorage.setItem('initialized', JSON.stringify(storedInitialized));
+
+		//Store modified accounts' data
+		let storedAccountsData = JSON.parse(window.localStorage.getItem('accounts'));
+		console.log(tempAccounts, length);
+		storedAccountsData.accounts = tempAccounts;
+		window.localStorage.setItem('accounts', JSON.stringify(storedAccountsData));
+
+		$.ajax({
```

Comment:
Would it be possible to use the SDK here? Seems like it's used elsewhere in app/services.
You may have a good reason not to use it though.

Reply:
Regardless, there should probably be some error checking of this call

Reply:
This should use the SDK. You may have to add the tokens API to the purecloud service. 

We do want to fire-and-forget this request though. This request just invalidates the token and then we remove it from our local cache. We won't stop that process if deleting the token fails, so there's no point in handling the response if it won't drive any action. May be worth adding a comment to mention this so future readers know it's intentional and not a bug.

Reply:
Here's the SDK function: https://developer.genesys.cloud/api/rest/client-libraries/javascript/TokensApi#deleteTokensMe